src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Publish.targets

```diff
@@ -770,6 +846,29 @@ Copyright (c) .NET Foundation. All rights reserved.
 
   </Target>
 
+  <!--
+    ============================================================
+                                        BundlePublishDirectory
+
+    Bundle the items in PreBundlePublishDir into one file in PublishDir
+    ============================================================
+    -->
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Build.Bundle" />
+
+  <Target Name="BundlePublishDirectory" 
+          Condition="'$(PublishSingleFile)' == 'true'"
+          Inputs="$(PreBundlePublishDir)\**\*.*"
+          Outputs="$(PublishDir)\$(AssemblyName)$(_NativeExecutableExtension)">
+    <PropertyGroup>
+      <BundleArgs>--source $(PreBundlePublishDir) </BundleArgs>
+      <BundleArgs>$(BundleArgs) --apphost $(AssemblyName)$(_NativeExecutableExtension)</BundleArgs>
+      <BundleArgs Condition="'$(IncludePdbInSingleFile)'=='true'">$(BundleArgs) --pdb</BundleArgs>
+      <BundleArgs>$(BundleArgs) -o $(PublishDir)</BundleArgs>
+    </PropertyGroup>
+    
+    <Exec Command="$(DOTNET_HOST_PATH) $(MicrosoftDotnetBundle) $(BundleArgs)" />
```

Comment:
SDK folks, is this the right property to use for launching `dotnet.exe`?

Reply:
CC: @nguerrera